server/controllers/incidents.js

```diff
@@ -113,42 +102,37 @@ module.exports = {
 
   // update an incident
   update(req, res) {
-    return Incident
-      .findById(req.params.id,
-        {
-          include: [{
-            model: Level,
-            attributes: ['name']
-          }, {
-            model: Status,
-            attributes: ['status']
-          }, {
-            model: Location,
-            attributes: ['name', 'centre', 'country']
-          }, {
-            model: User, 
-            attributes: ['name', 'imageUrl', 'email']
-          }]
-        })
-      .then(incident => {
-        if (!incident) {
-          return res.status(404).send({
+    let assignedRole = req.body.assignedRole
+    return Incident.findById(req.params.id, {
+      include: includes
+    })
+      .then(Incident => {
+        if (!Incident) {
+          res.status(404).send({
             message: 'Incident Not Found', status: "fail"
-          });
-        }
-        return incident
-          .update({
-            description: req.body.description || incident.description,
-            subject: req.body.subject || incident.subject,
-            dateOccurred: req.body.dateOccurred || incident.dateOccurred,
-            statusId: req.body.statusId || incident.statusId,
-            locationId: req.body.locationId || incident.locationId,
-            categoryId: req.body.categoryId || incident.categoryId,
-            assigneeId: req.body.assigneeId || incident.assigneeId,
-            levelId: req.body.levelId || incident.levelId
           })
-          .then(() => res.status(200).send({ data: incident, status: "success" }))
-          .catch(error => res.status(400).send(error));
+        }
+        return Incident.update({
+          description: req.body.description || Incident.description,
+          subject: req.body.subject || Incident.subject,
+          dateOccurred: req.body.dateOccurred || Incident.dateOccurred,
+          statusId: req.body.statusId || Incident.statusId,
+          locationId: req.body.locationId || Incident.locationId,
+          categoryId: req.body.categoryId || Incident.categoryId,
+          levelId: req.body.levelId || Incident.levelId
+        })
+      }).then(Incident => {
+        return User.findById(req.body.userId).then(assignees => {
+          Incident.addAssignees(assignees, { through: { assignedRole: 'assignee' }});
```

Comment:
I think we were supposed to remove the through part.

Reply:
We are meant to retain it because it updates the role of the assignee to either options.